mz_msisdn/__init__.py

```diff
@@ -1,39 +1,77 @@
-def mz_validator(number: str, nr_format: str = "international") -> list:
-    """ Verify if the given number is valid
-
-    Args:
-        number (str): The phone number to be validated
-        nr_format (str, optional): The number has international identification code. Defaults to international.
-
-    Returns:
-        list: return a list containing all errors identified. If the list is empty it means the number is valid.
-    """
-    number = number.strip()
-    network_codes = ["82", "83", "84", "85", "86", "87"]
-    inter_format = True if nr_format == "international" else False
-    number_size = 12 if inter_format else 9
-    error_messages = []
-
-    error_type = {
-        "size": f"The phone number must be :attr: characters long. currently with {len(number)}.",
-        "format": "The phone number must start with 258.",
-        "network_code": f"Invalid network code. Valid codes: {network_codes}."
+"""
+--------------------------------------------------------
+The Validator class.
+This library check numbers, based on the format of Mozambican numbers.
+--------------------------------------------------------
+"""
+
+class Validator:
+
+    __error_type = {
+        "size": "The phone number must be 9 or 12 characters long. currently with {}",
+        "format": "The phone number must start with {}.",
+        "network_code": "Invalid network code. {}."
+    }
+    
+    __error_messages = []
+
+    __codes =  {
+        "mz": {
+            "network_code": ["82", "83", "84", "85", "86", "87"],
+            "country_code": ["258"]
+        }
     }
+ 
+    def __init__(self, number, format = "mz"):
+        """
+            Args:
+            number (str): The phone number to be validated
+            format (str, optional): The number has international identification code. Defaults to international.
```

Comment:
The project was created to verify the number of the mobile network of mozambique, for that reason by default it checks for the mozambican mobile network.
For this reason I created the dictionary ** __ codes **.
```py
__codes = {
         "mz": {
             "network_code": ["82", "83", "84", "85", "86", "87"],
             "country_code": ["258"]
         }
```
In the future we will validate for different countries, and it will be very simple to include this feature.
To validate numbers for ** Kenya **, ** Angola ** and more, just include your data in ** __ codes **.
```py
__codes = {
         "mz": {
             "network_code": ["82", "83", "84", "85", "86", "87"],
             "country_code": ["258"]
         },
         "kn": {
             "network_code": ["...", "...", "..."],
             "country_code": ["254"]
         },
         "an": {
             "network_code": ["...", "..."],
             "country_code": ["244"]
         }
}
```

Finally, we would use the following syntax to validate numbers from Kenya or Angola.
```py
numberValidated = Validator("894214546","an")
numberValidated2 = Validator("312110129","kn")
print(numberValidated.get_errors())
print(numberValidated2.get_errors())
```